Data/ByteString.hs

```diff
@@ -1229,11 +1229,11 @@ group xs
 
 -- | The 'groupBy' function is the non-overloaded version of 'group'.
 groupBy :: (Word8 -> Word8 -> Bool) -> ByteString -> [ByteString]
-groupBy k xs = case uncons xs of
-  Nothing     -> []
-  Just (h, t) -> take n xs : groupBy k (drop n xs)
+groupBy k xs
+    | null xs   = []
+    | otherwise = unsafeTake n xs : groupBy k (unsafeDrop n xs)
     where
-        n = 1 + findIndexOrEnd (not . k h) t
+        n = 1 + findIndexOrEnd (not . k (unsafeHead xs)) (unsafeTail xs)
```

Comment:
Employing `uncons` was a right move, because it allowed us to avoid `unsafeHead` / `unsafeTail`, which is great. My comment was about `unsafeTake` / `unsafeDrop` only.